spec/master_manipulator/log_spec.rb

```diff
@@ -0,0 +1,128 @@
+    log_result = Beaker::Result.new('host', 'cmd')
+    log_result.stdout = log_dir
+    expect(dummy_class).to receive(:on).with(beaker_host, cmd_1).and_return(log_result)
+
+    date_result = Beaker::Result.new('host', 'cmd')
+    date_result.stdout = "1976-07-04"
+    expect(dummy_class).to receive(:on).with(beaker_host, cmd_date).and_return(date_result)
+  end
+  
+#add test for if log file is missing
+
+  context 'normal' do 
+      
+    describe '.rotate_puppet_server_log' do
+      it 'with correct required argument' do
+        shared_dsl_expectations
+
+        log_file_test_result = Beaker::Result.new('host', 'cmd')
+        log_file_test_result.exit_code = 0
+        expect(dummy_class).to receive(:on).with(beaker_host, cmd_test_log_file).and_return(log_file_test_result)
+
+        backup_log_file_test_result = Beaker::Result.new('host', 'cmd')
+        backup_log_file_test_result.exit_code = 1
+        expect(dummy_class).to receive(:on).with(beaker_host, /#{cmd_test_backup_log_file_0}/ ).and_return(backup_log_file_test_result)
+
+        copy_truncate_result = Beaker::Result.new('host', 'cmd')
+        copy_truncate_result.exit_code = 0
+        expect(dummy_class).to receive(:on).with(beaker_host,"cp #{log_file} #{backup_log_file_base}.0.log; cat /dev/null > #{log_file}").and_return(copy_truncate_result)
+
+        expect{dummy_class.rotate_puppet_server_log(beaker_host)}.not_to raise_error
+      end
+
+      it 'backup already exists' do
+        shared_dsl_expectations
+
+        log_file_test_result = Beaker::Result.new('host', 'cmd')
+        log_file_test_result.exit_code = 0
+        expect(dummy_class).to receive(:on).with(beaker_host, cmd_test_log_file).and_return(log_file_test_result)
+
+        backup_log_file_0_test_result = Beaker::Result.new('host', 'cmd')
+        backup_log_file_0_test_result.exit_code = 0
+        expect(dummy_class).to receive(:on).with(beaker_host, /#{cmd_test_backup_log_file_0}/ ).and_return(backup_log_file_0_test_result)
+      
+        backup_log_file_1_test_result = Beaker::Result.new('host', 'cmd')
+        backup_log_file_1_test_result.exit_code = 1
+        expect(dummy_class).to receive(:on).with(beaker_host, /#{cmd_test_backup_log_file_1}/ ).and_return(backup_log_file_1_test_result)
+
+        copy_truncate_result = Beaker::Result.new('host', 'cmd')
+        copy_truncate_result.exit_code = 0
+        expect(dummy_class).to receive(:on).with(beaker_host,"cp #{log_file} #{backup_log_file_base}.1.log; cat /dev/null > #{log_file}").and_return(copy_truncate_result)
+
+        expect{dummy_class.rotate_puppet_server_log(beaker_host)}.not_to raise_error
+      end
+
+      it 'with too many arguments' do
+        expect{ dummy_class.rotate_puppet_server_log(beaker_host, {}, 'No Bueno') }.to raise_error(ArgumentError)
+      end
+
+      it 'with no arguments' do
+        expect{ dummy_class.rotate_puppet_server_log }.to raise_error(ArgumentError)
+      end
+      
+      it 'Log missing' do
```

Comment:
Nit: Why use upper case here when all the other `it` statements are all lower case?